inst/shiny/map/server.R

```diff
@@ -1,46 +1,52 @@
 library(leaflet)
-
-# --- hack alert! ---- 
-# sites = unitted::v(mda.streams::get_meta(types = c("basic"), out = c('site_name')))$site_name
-# # do this expensive operation once
-# 
-# site_urls = data.frame(site_name=sites, sb_url=mda.streams::locate_site(sites, 'url', browser=FALSE))
-# save(site_urls, file = "../powstreams/inst/shiny/data/site_urls.RData")
-load(system.file('shiny','data','site_urls.RData',package='powstreams'))
-# /--- hack alert! ---- 
 server <- function(input, output, session) {
   
-  pal <- leaflet::colorQuantile("YlOrRd", NULL, n = 8)
   
-  points <- eventReactive(input$recalc, {
-    site_data = unitted::v(mda.streams::get_meta(types = c("basic"), out = c('site_name','long_name','lat','lon',input$variable)))
-    site_data = merge(site_data, site_urls)
-    site_data[rowSums(is.na(site_data)) == 0,]
-  }, ignoreNULL = FALSE)
+  
+  points <- reactive({
+    site_data = mda.streams::get_meta(out = c('site_name','long_name','lat','lon','sciencebase_id',input$variable))
+    site_units = unitted::get_units(site_data)
+    site_data = unitted::v(site_data)
+    list(data=site_data[rowSums(is.na(site_data)) == 0,], units=site_units)
+  })
   
   
   
   output$mymap <- leaflet::renderLeaflet({
-    data <- points()
+    pal <- leaflet::colorQuantile("YlOrRd", NULL, n = 7)
+    data_out <- points()
+    data = data_out$data
+    units = data_out$units
+    
+    ramp = tryCatch({
+      pal(data[[input$variable]])
+      pal
+    }, error = function(e) {
+      leaflet::colorBin("YlOrRd", NULL, n = 7)
+    })
     leaflet::leaflet() %>%
       leaflet::addProviderTiles("CartoDB.Positron",
                        options = leaflet::providerTileOptions(noWrap = TRUE)
       ) %>%
-      leaflet::addCircleMarkers(data = data, radius = 10, color=~pal(data[[input$variable]]), 
+      leaflet::addCircleMarkers(data = data, radius = 10, color=~ramp(data[[input$variable]]),
                        popup = ~paste0(long_name, '<br/>', 
-                                       sprintf("<a href='%s' target=_blank>%s</a>",sb_url,site_name), '<br/> value: ',data[[input$variable]]))
+                                      sprintf("<a href='https://www.sciencebase.gov/catalog/folder/%s' target=_blank>%s</a>",sciencebase_id,site_name), '<br/> ',
+                                      tail(strsplit(input$variable,'[.]')[[1]],1),': ',data[[input$variable]],' (',units[[input$variable]],')'))
```

Comment:
the units may not actually need parentheses here - consider "15 mins" vs "15 (mins)"